# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursor.cpp
# Commit: 13e00068cb2b
# Full Hash: 13e00068cb2bf553f6badfbacbbc7decfcfdf4bf
# Author: Birunthan Mohanathas <birunthan@mohanathas.com>
# Date: 2015-08-23 21:18:49
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Part 5: Send two records with every ObjectStoreCursorResponse. r=khuey
# ==============================================================================

diff -r 8e9fb56cc22a -r 13e00068cb2b dom/indexedDB/IDBCursor.cpp
--- a/dom/indexedDB/IDBCursor.cpp	Wed Aug 19 14:59:28 2015 -0700
+++ b/dom/indexedDB/IDBCursor.cpp	Wed Aug 19 14:59:29 2015 -0700
@@ -483,7 +483,7 @@
                  IDB_LOG_STRINGIFY(key));
   }
 
-  mBackgroundActor->SendContinueInternal(ContinueParams(key), Key());
+  mBackgroundActor->SendContinueInternal(ContinueParams(key), mKey);
 
   mContinueCalled = true;
 }
@@ -541,7 +541,7 @@
                  aCount);
   }
 
-  mBackgroundActor->SendContinueInternal(AdvanceParams(aCount), Key());
+  mBackgroundActor->SendContinueInternal(AdvanceParams(aCount), mKey);
 
   mContinueCalled = true;
 }
@@ -574,6 +574,8 @@
   MOZ_ASSERT(!mKey.IsUnset());
   MOZ_ASSERT_IF(mType == Type_Index, !mPrimaryKey.IsUnset());
 
+  mBackgroundActor->InvalidateCachedResponses();
+
   IDBObjectStore* objectStore;
   if (mType == Type_ObjectStore) {
     objectStore = mSourceObjectStore;
@@ -691,6 +693,8 @@
   MOZ_ASSERT(mType == Type_ObjectStore || mType == Type_Index);
   MOZ_ASSERT(!mKey.IsUnset());
 
+  mBackgroundActor->InvalidateCachedResponses();
+
   IDBObjectStore* objectStore;
   if (mType == Type_ObjectStore) {
     objectStore = mSourceObjectStore;